classify-split-extract-workflow/classify-job/config.py

```diff
@@ -0,0 +1,350 @@
+    Retrieves the document types configuration.
+
+    Returns:
+        Dict: The document types configuration.
+    """
+
+    return get_config("document_types_config")
+
+
+def get_parser_by_doc_type(doc_type: str) -> Optional[Dict]:
+    """
+    Retrieves the parser by document type.
+
+    Args:
+        doc_type (str): The document type.
+
+    Returns:
+        Optional[Dict]: The parser configuration.
+    """
+
+    doc = get_document_types_config().get(doc_type)
+    if not doc:
+        logger.error(f"doc_type {doc_type} not present in document_types_config")
+        return None
+
+    parser_name = doc.get("parser")
+    return get_parser_config().get(parser_name)
+
+
+def get_config(config_name: Optional[str] = None) -> Dict:
+    """
+    Retrieves the configuration data.
+
+    Args:
+        config_name (Optional[str]): The configuration name.
+
+    Returns:
+        Dict: The configuration data.
+    """
+
+    config_data_loaded = load_config(CONFIG_BUCKET, CONFIG_FILE_NAME)
+    assert config_data_loaded, f"Unable to locate '{config_name}' or incorrect JSON file"
+
+    if config_name:
+        config_item = config_data_loaded.get(config_name, {})
+    else:
+        config_item = config_data_loaded
+    return config_item
+
+
+def load_config(bucket_name: str, filename: str) -> Optional[Dict]:
+    """
+    Loads the configuration data from a GCS bucket or local file.
+
+    Args:
+        bucket_name (str): The GCS bucket name.
+        filename (str): The configuration file name.
+
+    Returns:
+        Optional[Dict]: The configuration data.
+    """
+
+    global bucket
+    if not bucket:
+        init_bucket(bucket_name, filename)
+
+    blob = bucket.get_blob(filename)
+    last_modified_time = blob.updated
+    global last_modified_time_of_object
+    global config_data
+    if last_modified_time == last_modified_time_of_object:
+        return config_data
+    else:
+        logger.info(f"load_config - Reloading config from: {filename}")
+        try:
+            if blob.exists():
+                config_data = json.loads(blob.download_as_text(encoding="utf-8"))
+                last_modified_time_of_object = last_modified_time
+                return config_data
+            else:
+                logger.error(f"load_config - Error: file does not exist gs://{bucket_name}/{filename}")
+        except Exception as e:
+            logger.error(f"load_config - Error: while obtaining file from GCS gs://{bucket_name}/{filename} {e}")
+            # Fall-back to local file
+            logger.warning(f"load_config - Warning: Using local {filename}")
+            with open(os.path.join(os.path.dirname(__file__), "config", filename)) as json_file:
+                config_data = json.load(json_file)
+                return config_data
+
+
+def get_docai_settings() -> Dict:
+    """
+    Retrieves the Document AI settings configuration.
+
+    Returns:
+        Dict: The Document AI settings configuration.
+    """
+
```

Comment:
I feel like quite a few of this methods could be combined/simplified. Maybe try plugging this into Gemini to see how it could be reduced?

Reply:
No good suggestions from neither GenAI code assistants. But I have spent a while trying to optimize and time limited the effort.